src/helpers/price-labels.ts

```diff
@@ -24,56 +25,31 @@ export const formatPrice = (
   return formatter.format(price);
 };
 
-export const getFrequencyLabel = (period: Period): string => {
-  const numberPeriods = period.number;
-  if (numberPeriods === 1) {
-    switch (period.unit) {
-      case PeriodUnit.Year:
-        return "yearly";
-      case PeriodUnit.Month:
-        return "monthly";
-      case PeriodUnit.Week:
-        return "weekly";
-      case PeriodUnit.Day:
-        return "daily";
-    }
-  } else {
-    return `every ${numberPeriods} ${period.unit}s`;
-  }
-};
-
-export const getLengthLabel = (period: Period): string => {
-  const numberPeriods = period.number;
-  if (numberPeriods === 1) {
-    switch (period.unit) {
-      case PeriodUnit.Year:
-        return "1 year";
-      case PeriodUnit.Month:
-        return "1 month";
-      case PeriodUnit.Week:
-        return "1 week";
-      case PeriodUnit.Day:
-        return "1 day";
-    }
-  } else {
-    return `${numberPeriods} ${period.unit}s`;
-  }
-};
-
-export const getRenewsLabel = (duration: string): string => {
+export const getTranslatedPeriodFrequency = (
+  duration: string,
+  translator: Translator,
+): string => {
   const period = parseISODuration(duration);
   if (!period) {
     return "unknown";
```

Comment:
Shouldn't normally happen but does this need to be localized as well?

Reply:
Yes!

Good catch, I'll fix it

Reply:
Translated it